packages/react-flex/Spec.md

```diff
@@ -0,0 +1,540 @@
+
+#### _Deprecating_
+
+| Original Component | Name            | Comments                                               |
+| ------------------ | --------------- | ------------------------------------------------------ |
+| StackItem          | `disableShrink` | Extraneous prop given the existance of `shrink`.       |
+| StackItem          | `verticalFill`  | Redundant. Same as a style override of `height: 100%`. |
+| FlexItem           | `size`          | Renamed to `basis`.                                    |
+
+## Sample Code
+
+### Basic
+
+```HTML
+<Flex>
+ <span>1</span>
+ <span>2</span>
+</Flex>
+```
+
+### Direction
+
+```HTML
+<Flex direction="column-reverse">
+  <span>bottom</span>
+  <span>top</span>
+</Flex>
+```
+
+### Alignment
+
+```HTML
+<Flex verticalAlign="center">
+  <span>vertically centered</span>
+  <span>vertically centered</span>
+</Flex>
+```
+
+```HTML
+<Flex horizontalAlign="center">
+  <span>horizontally centered</span>
+  <span>horizontally centered</span>
+</Flex>
+```
+
+### Spacing
+
+```HTML
+<Flex gap="5rem">
+  <span>5rem around me</span>
+  <span>5rem around me</span>
+</Flex>
+```
+
+### Wrapping
+
+```HTML
+<Flex wrap>
+  <span>1</span>
+  <span>2</span>
+</Flex>
+```
+
+### Order
+
+```HTML
+<Flex>
+  <span>2</span>
+  <span order={1}>1</span>
+</Flex>
+```
+
+### Growing
+
+#### Globally
+
+```HTML
+<Flex grow>
+  <span>wide</span>
+  <span>wide</span>
+</Flex>
+```
+
+#### Individually
+
+```HTML
+<Flex>
+  <span>thin</span>
+  <span grow>wide</span>
+</Flex>
+```
+
+### Shrinking
+
+#### Globally
+
+```HTML
+<Flex shrink={false}>
+  <span>wide</span>
+  <span>wide</span>
+</Flex>
+```
+
+#### Individually
+
+```HTML
+<Flex>
+  <span>thin</span>
+  <span shrink={false}>wide</span>
+</Flex>
+```
+
+### Complex example
+
+```HTML
+<Flex
+  as="section"
+  direction="row-reverse"
+  horizontalAlign="space-between"
+  verticalAlign="center"
+  gap="5rem"
+  wrap grow inline>
+  <span order={3}>3</span>
+  <span shrink={false}>1</span>
+  <span basis="25%">2</span>
+</Flex>
+```
+
+## Migration
+
+_Describe what will need to be done to upgrade from the existing implementations:_
+
+### _Migration from v8_
+
+#### Stack
+
+##### `disableShrink`
+
+Renamed from `disableShrink` to `shrink`
+
+Before
+
+```HTML
+<Stack disableShrink>
+ items
+</Stack>
+```
+
+After
+
+```HTML
+<Flex shrink="0">
+  items
+</Flex>
+```
+
+```HTML
+<Flex shrink={false}>
+  items
+</Flex>
+```
+
+##### `horizontal`
+
+Before
+
+```HTML
+<Stack horizontal>
+ items
+</Stack>
+```
+
+After
+
+```HTML
+<Flex>
+  items
+</Flex>
+```
+
+##### `childrenGap`
+
+Renamed from `childrenGap` to `gap`
+
+##### `reversed` [DEPRECATED]
+
+Before
+
+```HTML
+<Stack reversed>
+ items
+</Stack>
+```
+
+Alternative
+
+```HTML
+<Flex direction="column-reverse">
+  items
+</Flex>
+```
+
+##### `maxHeight` [DEPRECATED]
+
+Before
+
+```HTML
+<Stack maxHeight="30px">
+ items
+</Stack>
+```
+
+Alternative
+
+```HTML
+<Flex style={{maxHeight:"30px"}}>
```

Comment:
While this should be possible I don't think we should go straight for inline styles since they break overrides for most applications, how about we use simple CSS or use our styling solution ? 

```suggestion

const maxHeightClass = useStyles(); 

<Flex className={maxHeightClass}>
```

Reply:
Sorry, this was my proposal while being completely unaware of our current styling system! I will look into what we have right now and update the examples to be closer to our implementation. Thanks!

Reply:
> While this should be possible I don't think we should go straight for inline styles since they break overrides for most applications, 

Can you elaborate pls? Are we gonna restrict users to use standard DOM properties ? 
> Note: inline styles are definitely antipattern, we should be explicit what will be supported though :)

Reply:
All native prop should be supported according to our current practices. But we should not enourage inline styling.

Reply:
> But we should not enourage inline styling.

agree.

Note: from my experience, if we don't ship lint rules for that kind of encouragement/following best practices , we are pretty much powerless on how consumers are gonna use it.

Reply:
To be clear, my suggestion was to stay away from inline styles on our end (we control that I hope 😄 ). We do know that inline styles should be avoided and it's not a major change to the examples

Reply:
This is just the migration section for the spec and not the actual migration guide. The idea is to simply show how to get the exact same behaviour as using the prop in the old component and shouldn't be taken as a best practice. This will also be made more user friendly when writing the actual migration guide.
Regardless, do you think using plain CSS would be better? It's more neutral to the ecosystem and no changes would be needed should the styling system change. That or informing that it shouldn't be taken as the best practice but is instead a react implementation on how you could do it.